apps/unlock/lib/controller.ex

```diff
@@ -103,10 +112,68 @@ defmodule Unlock.Controller do
     |> send_resp(response.status, response.body)
   end
 
-  defp process_resource(conn, %Unlock.Config.Item.SIRI{}) do
+  defp process_resource(conn, %Unlock.Config.Item.GTFS.RT{}), do: send_not_allowed(conn)
+
+  # NOTE: this code is designed for private use for now. I have tracked
+  # what is required or useful for public opening later here:
+  # https://github.com/etalab/transport-site/issues/2476
+  defp process_resource(%{method: "POST"} = conn, %Unlock.Config.Item.SIRI{} = item) do
+    {:ok, body, conn} = Plug.Conn.read_body(conn, length: 1_000_000)
+
+    parsed = Unlock.SIRI.parse_incoming(body)
+
+    {modified_xml, external_requestor_refs} =
+      Unlock.SIRI.RequestorRefReplacer.replace_requestor_ref(parsed, item.requestor_ref)
+
+    # NOTE: here we assert both that the requestor ref is what is expected, but also that it
+    # is met once only. I am not deduping them at the moment on purpose, maybe we'll do that
+    # later based on experience.
+    if external_requestor_refs == [Application.fetch_env!(:unlock, :siri_public_requestor_ref)] do
+      handle_authorized_siri_call(conn, item, modified_xml)
+    else
+      send_resp(conn, 403, "Forbidden")
+    end
+  end
+
+  defp process_resource(%{method: "GET"} = conn, %Unlock.Config.Item.SIRI{}),
+    do: send_not_allowed(conn)
+
+  defp send_not_allowed(conn) do
     conn
-    |> put_status(501)
-    |> text("Not Implemented")
+    |> send_resp(405, "Method Not Allowed")
+  end
+
+  @spec handle_authorized_siri_call(Plug.Conn.t(), Unlock.Config.Item.SIRI.t(), Saxy.XML.element()) :: Plug.Conn.t()
+  defp handle_authorized_siri_call(conn, %Unlock.Config.Item.SIRI{} = item, xml) do
+    body = Saxy.encode_to_iodata!(xml)
```

Comment:
```suggestion
    body = Saxy.encode_to_iodata!(xml, [version: "1.0"])
```

Faut peut-être rajouter notre propre prolog au moment de faire l'appel au producteur ?

Reply:
Oui tu as tout bien compris @fchabouis !

Et compte tenu de :

- notre discussion tous les deux hier,
- le fait que XML 1.1 est assez différent de 1.0 (https://www.w3.org/TR/2006/REC-xml11-20060816/#sec-xml11) et qu'on ne veut surtout pas aller sur ce terrain là,
- le fait qu'on veut "cercler" le comportement et réduire les ambiguïtés (surtout avec N serveurs différents en face, certains peut-être assez legacy) pour éviter tout problème,

Je commit ton changement, et j'adapte les tests en fonction, en essayant d'avoir un test parlant (pas de prolog en entrée, un prolog en sortie etc)